src/pyvesync/vesyncbasedevice.py

```diff
@@ -11,55 +12,59 @@ class VeSyncBaseDevice(object):
+            self.device_name = details.get('deviceName', None)
+            self.device_image = details.get('deviceImg', None)
+            self.cid = details.get('cid', None)
+            self.device_status = details.get('deviceStatus', None)
+            self.connection_status = details.get('connectionStatus', None)
+            self.connection_type = details.get('connectionType', None)
+            self.device_type = details.get('deviceType', None)
+            self.type = details.get('type', None)
+            self.uuid = details.get('uuid', None)
+            self.config_module = details.get('configModule', None)
+            self.mac_id = details.get('macID', None)
+            self.mode = details.get('mode', None)
+            self.speed = details.get('speed', None)
+            self.extension = details.get('extension', None)
+            self.current_firm_version = details.get('currentFirmVersion', None)
 
-        self.device_name = None
-        self.device_image = None
-        self.cid = None
-        self.device_status = None
-        self.connection_status = None
-        self.connection_type = None
-        self.device_type = None
-        self.type = None
-        self.uuid = None
-        self.config_module = None
-        self.mac_id = None
-        self.mode = None
-        self.speed = None
-        self.extension = None
-        self.current_firm_version = None
+        else:
+            logger.error('No cid found for ' + self.__class__.__name__)
 
-        self.configure(details)
+    def __eq__(self, other):
+        return other.cid == self.cid
 
-    def configure(self, details):
-        self.device_name = details.get('deviceName', None)
-        self.device_image = details.get('deviceImg', None)
-        self.cid = details.get('cid', None)
-        self.device_status = details.get('deviceStatus', None)
-        self.connection_status = details.get('connectionStatus', None)
-        self.connection_type = details.get('connectionType', None)
-        self.device_type = details.get('deviceType', None)
-        self.type = details.get('type', None)
-        self.uuid = details.get('uuid', None)
-        self.config_module = details.get('configModule', None)
-        self.mac_id = details.get('macID', None)
-        self.mode = details.get('mode', None)
-        self.speed = details.get('speed', None)
-        self.extension = details.get('extension', None)
-        self.current_firm_version = details.get('currentFirmVersion', None)
+    def __hash__(self):
+        return hash(self.cid)
 
-        if self.cid is None:
-            logger.error('Error getting cid for ' + self.__class__.__name__)
+    def __str__(self):
+        return "Device Name: {}, Device Type: {}, Status: {}".format(
+            self.device_name, self.device_type, self.device_status)
 
-    def is_on(self) -> str:
+    def __repr__(self):
+        return "DevClass: {}, Name:{}, DevStatus: {}, CID: {}" \
```

Comment:
What is DevClass?

Reply:
`__repr__` is only used for consoles and debugging to give a string representation of devices.  The `DevClass` shows the class the device was instantiated on.  The `__repr__` function allows the object to return a unique readable string if inside of a list.  I used it to debug the device lists in the vesync object.  It shouldn't have an impact on someone just using a script to access the api.